app/routes.py

```diff
@@ -1,2 +1,233 @@
+def create_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return make_response(
+            {"details": f"Invalid data"}, 400
+        )
+
+    new_goal = Goal(
+        title = request_body["title"]
+    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response(
+        {"goal":(new_goal.to_dict())}, 201
+    )
+
+# reads all created Goals
+@goals_bp.route("", methods=["GET"], strict_slashes=False)
+def get_goals():
+    goals_response = []
+
+    goals = Goal.query.all()
+
+    for goal in goals:
+        goals_response.append(goal.to_dict())
+    return jsonify(goals_response), 200
+
+# single Task: Goal
+@goals_bp.route("/<goal_id>", methods=["GET"], strict_slashes=False)
+def get_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+
+    return make_response(
+        {"goal":(goal.to_dict())}, 200
+    )
+
+# single Goal: update
+@goals_bp.route("/<goal_id>", methods=["PUT"], strict_slashes=False)
+def update_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return make_response(
+            {"details": f"Invalid data"}, 400
+        )
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    return make_response(
+        {"goal":(goal.to_dict())}, 200
+    )
+
+# single Goal: delete
+@goals_bp.route("/<goal_id>", methods=["DELETE"], strict_slashes=False)
+def delete_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response(
+        {"details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"}, 200
+    )
+
+# send a list of Task IDs to a Goal
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"], strict_slashes=False)
+def post_tasks_to_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    request_body = request.get_json()
+    task_ids = request_body["task_ids"]
+
+    # loops through task_ids list, which are the IDs of Tasks that client want to associate with this specific goal 
+    for task in task_ids: 
+        # uses each task_id to query the Task object that needs to be updated with match_goal_id foreign key 
+        task_to_add_to_goal = Task.query.get_or_404(task)
+        # uses goal_id from Goal queried above, assigns value to the goal_id foreign key of each task to add
+        # loops through given task_ids until each task in the post request has been updated
+        task_to_add_to_goal.match_goal_id = goal.goal_id 
+        
+    db.session.commit()
+
+    return make_response({"id": goal.goal_id, "task_ids": task_ids}, 200)
+
+# getting Tasks of one Goal
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"], strict_slashes=False)
+def read_tasks_for_goal(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    goal_and_tasks = []
+
+    tasks = Task.query.filter_by(match_goal_id=goal_id)
```

Comment:
In models/goal.py line 7 `tasks = db.relationship("Task", back_populates="goal")` is effectively doing the same thing as this query.  Using db.relationship creates a field `tasks` that is a list of the Task objects that have have the matching `goal_id` as their foreign key `match_goal_id`.